crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -342,290 +405,135 @@ pub struct PipelineCache {
     layout_cache: LayoutCache,
     shader_cache: ShaderCache,
     device: RenderDevice,
-    pipelines: Vec<CachedPipeline>,
-    waiting_pipelines: HashSet<CachedPipelineId>,
-    new_pipelines: Mutex<Vec<CachedPipeline>>,
+    render_pipeline_cache:
+        PipelineCacheInternal<RenderPipelineId, RenderPipelineDescriptor, RenderPipeline>,
+    compute_pipeline_cache:
+        PipelineCacheInternal<ComputePipelineId, ComputePipelineDescriptor, ComputePipeline>,
 }
 
 impl PipelineCache {
-    pub fn pipelines(&self) -> impl Iterator<Item = &CachedPipeline> {
-        self.pipelines.iter()
-    }
-
```

Comment:
I have removed this for now, since `RenderPipeline`s and `ComputePipeline`s are now stored separately. 
Not sure if we want to add a separate iterator for each, or just remove this method entirely.